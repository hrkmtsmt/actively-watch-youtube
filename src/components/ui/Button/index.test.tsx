import React from 'react';
import { describe, test, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { Button } from '.';
import '@testing-library/jest-dom/vitest';

describe('Button component', () => {
  test('snapshot', () => {
    render(<Button>Snapshot</Button>);
    expect(document.body.getElementsByTagName('button')[0]).toMatchSnapshot();
  });

  test('render dom with children.', () => {
    render(<Button>Hello</Button>);
    expect(screen.getByRole('button')).toHaveTextContent('Hello');
  });
});
