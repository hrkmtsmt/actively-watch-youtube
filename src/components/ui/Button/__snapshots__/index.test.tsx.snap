// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > snapshot 1`] = `
<button
  class="style_button__zqucfs0"
  type="button"
>
  Snapshot
</button>
`;
